internal/driver/macmapper_test.go

```diff
@@ -178,3 +182,233 @@ func TestMatchEndpointRefAddressToMAC(t *testing.T) {
 		})
 	}
 }
+
+func TestMACAddressMapper_UpdateMappings(t *testing.T) {
+	tests := []struct {
+		name              string
+		currentMap        map[string]string
+		expected          map[string]string
+		alternateExpected map[string]string
+	}{
+		{
+			name: "no update",
+			currentMap: map[string]string{
+				"creds1": "AA:BB:CC:DD:EE:FF",
+				"creds2": "11:22:33:44:55:66",
+			},
+			expected: map[string]string{
+				"aa:bb:cc:dd:ee:ff": "creds1",
+				"11:22:33:44:55:66": "creds2",
+			},
+		},
+		{
+			name: "single update",
+			currentMap: map[string]string{
+				"creds1": "aa:bb:cc:dd:ee:ff",
+				"creds2": "11:22:33:44:55:66",
+				"creds3": "ff:ee:dd:cc:bb:aa",
+			},
+			expected: map[string]string{
+				"aa:bb:cc:dd:ee:ff": "creds1",
+				"11:22:33:44:55:66": "creds2",
+				"ff:ee:dd:cc:bb:aa": "creds3",
+			},
+		},
+		{
+			name: "Add invalid macs",
+			currentMap: map[string]string{
+				"creds3": "FF:EE:DD:CC:BB:AA,asbc,asdf",
+				"creds1": "AA:BB:CC:DD:EE:FF",
+				"creds2": "11:22:33:44:55:66",
+			},
+			expected: map[string]string{
+				"aa:bb:cc:dd:ee:ff": "creds1",
+				"11:22:33:44:55:66": "creds2",
+				"ff:ee:dd:cc:bb:aa": "creds3",
+			},
+		},
+		{
+			name: "invalid macs",
+			currentMap: map[string]string{
+				"creds3": "FF:EE:DD:CC:BB:AA,asbc,asdf",
+				"creds1": "AA:BB:CC:DD:EE:FF",
+				"creds2": "11:22:33:44:55:66",
+			},
+			expected: map[string]string{
+				"aa:bb:cc:dd:ee:ff": "creds1",
+				"11:22:33:44:55:66": "creds2",
+				"ff:ee:dd:cc:bb:aa": "creds3",
+			},
+		},
+		{
+			name: "duplicate macs",
+			currentMap: map[string]string{
+				"creds1": "FF:EE:DD:CC:BB:AA",
+				"creds2": "11:22:33:44:55:66",
+				"creds3": "FF:EE:DD:CC:BB:AA",
+			},
+			expected: map[string]string{
+				"ff:ee:dd:cc:bb:aa": "creds1",
+				"11:22:33:44:55:66": "creds2",
+			},
+			alternateExpected: map[string]string{
+				"ff:ee:dd:cc:bb:aa": "creds3",
+				"11:22:33:44:55:66": "creds2",
+			},
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+
+			driver, mockService := createDriverWithMockService()
+			driver.macAddressMapper = NewMACAddressMapper(mockService)
+			driver.macAddressMapper.credsMap = test.currentMap
+			mockSecretProvider := &mocks.SecretProvider{}
+			mockLoggingClient := logger.MockLogger{}
+
+			if len(test.currentMap) == 0 {
+				return
+			}
+
+			for secretPath := range test.currentMap {
+				if strings.ToLower(secretPath) != noAuthSecretPath {
+					mockSecretProvider.On("GetSecret", secretPath, UsernameKey, PasswordKey, AuthModeKey).
+						Return(nil, nil)
+				}
+			}
+
+			mockService.On("GetSecretProvider").
+				Return(mockSecretProvider)
+			mockService.On("GetLoggingClient").Return(mockLoggingClient)
+			driver.macAddressMapper.UpdateMappings(test.currentMap)
+
+			if test.name == "duplicate macs" {
```

Comment:
testing via the name is finicky. use the existance of the `test.alternateExpected` field to determine whether or not to enter this code block.